### PR TITLE
API Version Extension

### PR DESCRIPTION
**Pull request by scisco**
**Related Issue(s):** #
N/A

**Proposed Changes:**

1. Adds a Version Extension to the API spec

**PR Checklist:**

- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [x] API only: I have run `npm run generate-all` to update the [generated OpenAPI files](https://github.com/radiantearth/stac-spec/blob/dev/api-spec/README.md#openapi-definitions).

### CODE REVIEW

File: api-spec/STAC-extensions.yaml [hunk 1]
```diff
@@ -405,6 +405,66 @@ paths:
...
+      description: returns the requested version of a collection
+      tags:
+        - Version Extension
+      parameters:
+        - $ref: '#/components/parameters/collectionId'
+        - $ref: '#/components/parameters/versionId'
+      responses:
+        '200':
+          $ref: '#/components/responses/Collections'
+  '/collections/{collectionId}/items/{featureId}/versions/':
```
Comment (philvarner) [inline api-spec/STAC-extensions.yaml]: extra trailing slash here

File: api-spec/extensions/version/README.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
+# Version API Extension
+
+**Extension [Maturity Classification](../../../extensions/README.md#extension-maturity): Pilot**
+
+The core API doesn't support semantics to creating and accessing different versions of an item or collection record. This version API extension defines the API resources and semantics for creating and accessing versioned records.
```
Comment (philvarner) [inline api-spec/extensions/version/README.md]: "item or collection record" -> "Item or Collection".

**Comment by gsf-sellis** [comment]
In the item-spec.md, we reference this page for possible link rel's: https://www.iana.org/assignments/link-relations/link-relations.xhtml

There you can find numerous link rel's related to versioning, e.g. latest-version, predecessor-version, successor-version, canonical, and version-history. We may want to consider using these link rel's in our versioning API.

File: api-spec/extensions/version/README.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
+# Version API Extension
+
+**Extension [Maturity Classification](../../../extensions/README.md#extension-maturity): Pilot**
```
Comment (m-mohr) [inline api-spec/extensions/version/README.md]: Maturity Level is probably Proposal or is there an implementation?

File: api-spec/extensions/version/README.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+| `GET /collections/{collectionID}/items/{featureId}/versions/`              | `application/json`  | Returns a catalog response with links to all versions of a given item. |
+| `GET /collections/{collectionID}/items/{featureId}/versions/{versionId}`              | `application/json`  | Returns an item record. |
+
+## How It Works
+
+When this extension is implemented, the API supports `versions` resource that list all versions of an item or collection and provides permanent links to each version.
+
+The latest version of an item is accessible at `/collections/{collectionID}/items/{itemsId}`. The record has a link with `"rel": "permalink"` that links to the permanent location of the record which is `/collections/{collectionID}/items/{itemsId}/versions/{versionId}`.
+
+If the record has a previous version it also provides a link to the permanent location of that version using `"rel": "previous_version"`
```
Comment (m-mohr) [inline api-spec/extensions/version/README.md]: The registered rel type `predecessor-version` would be better, I think. Would also better fit with the relation type successor-version, which is used with the deprecated flag.

File: api-spec/extensions/version/README.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        {
+            "rel" : "permalink",
+            "href": "/collections/my_collection/items/this_is_my_id/versions/02"
+        },
+        {
+            "rel": "previous_version",
+            "href": "/collections/my_collection/items/this_is_my_id/versions/01"
+        },
+        {
+            "rel": "latest_version",
```
Comment (m-mohr) [inline api-spec/extensions/version/README.md]: This rel type should better be: `latest-version`

File: api-spec/extensions/version/README.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+            "href": "/collections/my_collection/items/this_is_my_id"
+        },
+        {
+            "rel": "versions",
```
Comment (m-mohr) [inline api-spec/extensions/version/README.md]: This rel type should better be `version-history`.

**Comment by scisco** [comment]
@philvarner @m-mohr @gsf-sellis addressed your comments

**Comment by gsf-sellis** [comment]
Have we considered adding the successor-version link as described in RFC5829?

File: api-spec/extensions/version/README.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+## FAQ
+
+**How do I find the latest version of an item?**
+
+By going to `/collections/{collectionID}/items/{itemsId}` or by looking at `"rel": "latest-version"`.
+
+** How do I find the permalink of an item I'm looking at?**
+
+By looking at the `href` value of a link with `"rel": "permanlink"`
```
Comment (philvarner) [inline api-spec/extensions/version/README.md]: perma**n**link -> permalink

**Comment by m-mohr** [comment]
I'm thinking about how we can name the two Version Extensions be named better so that they don't have the same name. 

For API: Dynamic Item(?) Versioning Extension
For Core: Version Extension or Version and Deprecation Extension or File Version Extension?
Best practices: Static Item(?) Versioning Best Practices

What do you think @scisco ?

File: api-spec/extensions/version/README.md [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+        {
+            "rel": "predecessor-version",
+            "href": "/collections/my_collection/items/this_is_my_id/versions/01"
+        },
+        {
+            "rel": "latest-version",
+            "href": "/collections/my_collection/items/this_is_my_id"
+        },
+        {
+            "rel": version-history,
```
Comment (philvarner) [inline api-spec/extensions/version/README.md]: minor issue -- there are two examples of this without quotes

**Pull request merged**